modules/core/src/main/java/org/apache/ignite/IgniteCluster.java

```diff
@@ -442,17 +443,36 @@
      * Checks Ignite grid is active or not active.
      *
      * @return {@code True} if grid is active. {@code False} If grid is not active.
+     * @deprecated Use {@link #state()} instead.
      */
+    @Deprecated
     public boolean active();
 
     /**
      * Changes Ignite grid state to active or inactive.
      *
      * @param active If {@code True} start activation process. If {@code False} start deactivation process.
      * @throws IgniteException If there is an already started transaction or lock in the same thread.
+     * @deprecated Use {@link #state(ClusterState)} instead.
      */
+    @Deprecated
     public void active(boolean active);
 
+    /**
+     * Gets current cluster state.
+     *
+     * @return Current cluster state.
+     */
+    public ClusterState state();
+
+    /**
+     * Changes current cluster state to given {@code newState} cluster state.
+     *
+     * @param newState New cluster state.
+     * @throws IgniteException If there is an already started transaction or lock in the same thread.
+     */
```

Comment:
i don`t like api with exception throwing here, can we wait all tx commitments and change state ? its my own opinion if committee ok with it i`m close my mouth.

Reply:
If exceptions is only due to heterogeneous cluster its ok for me, but what about tx -we need to wail long tx and change state.

Reply:
We have the same behavior for `active(boolean)` command. 
This comment was copied from `active(boolean)` method and means that you can't change the cluster state inside transaction. See `ClusterReadOnlyModeSelfTest#testEnableReadOnlyModeInsideTransaction`